app/src/main/java/de/danoeh/antennapod/dialog/IntraFeedSortDialog.java

```diff
@@ -16,35 +16,59 @@
     @NonNull
     protected Context context;
 
-    public IntraFeedSortDialog(@NonNull Context context, @Nullable SortOrder sortOrder) {
+    final private String[] sortItems;
+    final private SortOrder[] sortValues;
+
+    public IntraFeedSortDialog(@NonNull Context context, @Nullable SortOrder sortOrder, @NonNull boolean isLocalFeed) {
         this.context = context;
         this.currentSortOrder = sortOrder;
+
+        if (isLocalFeed) {
+            sortItems = context.getResources().getStringArray(R.array.local_feed_episodes_sort_options);
+            final String[] localSortStringValues =
+                    context.getResources().getStringArray(R.array.local_feed_episodes_sort_values);
+            sortValues = SortOrder.valuesOf(localSortStringValues);
+        } else {
+            sortItems = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
+            final String[] commonSortStringValues =
+                    context.getResources().getStringArray(R.array.feed_episodes_sort_values);
+            sortValues = SortOrder.valuesOf(commonSortStringValues);
+        }
     }
 
     public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
-        final String[] valueStrs = context.getResources().getStringArray(R.array.feed_episodes_sort_values);
-        final SortOrder[] values = new SortOrder[valueStrs.length];
-        for (int i = 0; i < valueStrs.length; i++) {
-            values[i] = SortOrder.valueOf(valueStrs[i]);
-        }
+        final SortOrder[] values = sortValues;
+        final String [] items = sortItems;
```

Comment:
Why do you need to copy the arrays here?

Reply:
I don't, it was leftover from the previous implementation. I should have paid more attention, thanks!